test/box/gh-5958-rolled-back-replace-background-built-index.test.lua

```diff
@@ -0,0 +1,268 @@
+env = require('test_run')
+test_run = env.new()
+fiber = require('fiber')
+
+-- Create table with enough tuples to make index build/alter in background.
+s = box.schema.space.create('test', {engine = 'memtx'})
+i = s:create_index('pk',  {type='tree', parts={{1, 'uint'}}})
+for i = 1, 6000 do s:replace{i, i, "valid"} end
+
+started = false
+
+-- Array with origin tuples
+origin = {[0] = s:get{0}, [1] = s:get{1}, [2] = s:get{2}}
+
+
+
+------ Suite to test rolled back changes in build-in-background secondary index (unique and non-unique) ------
+
+
+
+test_run:cmd("setopt delimiter ';'");
+
+-- Make fiber joinable.
+function joinable(fib)
+    fib:set_joinable(true)
+    return fib
+end;
+
+-- Wait for fiber yield during building of index and then make all possible replaces:
+-- 1. replace nil -> x
+-- 2. replace x -> y
+-- 3. replace x -> nil
+-- and then roll them back.
+function disturb()
+    while not started do fiber.sleep(0) end
+    box.begin()
+    s:replace{0, 0, "invalid"}
+    s:replace{1, 1, "changed D:"}
+    i:delete{2}
+    box.rollback()
+end;
+
+-- Build secondary index in background.
+-- If the index will not be built in background, test will not be passed
+-- because it will run out of time (disturber:join() will wait forever).
+function create(is_unique)
+    started = true
+    s:create_index('sk', {unique=is_unique, type='tree', parts={{2, 'uint'}}})
+    started = false
+end;
+
+test_run:cmd("setopt delimiter ''");
+
+disturber = joinable(fiber.new(disturb))
+creator = joinable(fiber.new(create, false))
+
+disturber:join()
+creator:join()
+
+-- Check if all changes were rolled back in all indexes.
+assert(s.index.pk:get{0} == origin[0] and s.index.pk:get{0} == s.index.sk:select{0}[1])
```

Comment:
Let's put these checks in a separate function - they are in fact the same in two cases.

Reply:
Moreover, you can even put the whole two cases in one function with is_unique arg.